module/dkg/client.go

```diff
@@ -0,0 +1,218 @@
+	latestBlock, err := c.flowClient.GetLatestBlock(ctx, true)
+	if err != nil {
+		return fmt.Errorf("could not get latest block from node: %w", err)
+	}
+
+	tx := sdk.NewTransaction().
+		SetScript(templates.GenerateSendDKGWhiteboardMessageScript(c.env)).
+		SetGasLimit(9999).
+		SetReferenceBlockID(latestBlock.ID).
+		SetProposalKey(c.accountAddress, int(c.accountKeyIndex), account.Keys[int(c.accountKeyIndex)].SequenceNumber).
+		SetPayer(c.accountAddress).
+		AddAuthorizer(c.accountAddress)
+
+	err = tx.AddArgument(cadence.NewString(string(msg.Data)))
+	if err != nil {
+		return fmt.Errorf("could not add whiteboard dkg message to transaction: %v", err)
+	}
+
+	// sign payload using account signer
+	err = tx.SignPayload(c.accountAddress, int(c.accountKeyIndex), c.signer)
+	if err != nil {
+		return fmt.Errorf("could not sign transaction: %w", err)
+	}
+
+	// submit signed transaction to node
+	txID, err := c.submitTx(tx)
+	if err != nil {
+		return fmt.Errorf("failed to submit transaction: %w", err)
+	}
+
+	// wait for transaction to be sealed
+	result := &sdk.TransactionResult{Status: sdk.TransactionStatusUnknown}
+	for result.Status != sdk.TransactionStatusSealed {
+		result, err = c.flowClient.GetTransactionResult(ctx, txID)
+		if err != nil {
+			return fmt.Errorf("could not get transaction result: %w", err)
+		}
+
+		// if the transaction has expired we skip waiting for seal
+		if result.Status == sdk.TransactionStatusExpired {
+			return fmt.Errorf("broadcast dkg message transaction has expired")
+		}
+
+		// wait 1 second before trying again.
+		time.Sleep(time.Second)
+	}
+
+	return nil
+}
+
+// ReadBroadcast reads the broadcast messages from the smart contract.
+// Messages are returned in the order in which they were broadcast (received
+// and stored in the smart contract)
+func (c *Client) ReadBroadcast(fromIndex uint, referenceBlock flow.Identifier) ([]messages.DKGMessage, error) {
+
+	ctx := context.Background()
+
+	template := templates.GenerateGetDKGLatestWhiteBoardMessagesScript(c.env)
+	dkgMessages, err := c.flowClient.ExecuteScriptAtBlockID(ctx, sdk.Identifier(referenceBlock), template, []cadence.Value{cadence.NewInt(int(fromIndex))})
+	if err != nil {
+		return nil, fmt.Errorf("could not execute read broadcast script")
+	}
+
+	// TODO: convert from `cadence.Array` to `[]messages.DKGMessage`
+	return dkgMessages.ToGoValue().([]messages.DKGMessage), nil
+}
+
+// SubmitResult submits the final public result of the DKG protocol. This
+// represents the group public key and the node's local computation of the
+// public keys for each DKG participant
+func (c *Client) SubmitResult(groupPublicKey crypto.PublicKey, publicKeys []crypto.PublicKey) error {
+
+	ctx := context.Background()
+
+	// get account for given address
+	account, err := c.flowClient.GetAccount(ctx, c.accountAddress, grpc.EmptyCallOption{})
+	if err != nil {
+		return fmt.Errorf("could not get account details: %v", err)
```

Comment:
```suggestion
		return fmt.Errorf("could not get account details: %w", err)
```

Reply:
`%v` doesn't wrap the child error, so it will break error type checking with `errors.As` and `errors.Is` -- you pretty much always want to use `%w` for errors.